tests/graph/test_vector_network_wrapper.py

```diff
@@ -0,0 +1,166 @@
+        ]
+
+    def test_parse_ini_value(self, _config_fixture):
+        v = VectorNetworkWrapper(_config_fixture)
+        assert v._parse_ini_value("a,b,c") == ["a", "b", "c"]
+        assert v._parse_ini_value("abc") == "abc"
+        assert v._parse_ini_value("") is None
+
+    def test_parse_ini_filename(self, _config_fixture):
+        v = VectorNetworkWrapper(_config_fixture)
+        file_paths = v._parse_ini_filename("_test_lines.geojson, test.geojson")
+        assert file_paths[0].is_file()
+        assert not file_paths[1].is_file()
+
+    def test_setup_global(self, _config_fixture):
+        v = VectorNetworkWrapper(_config_fixture)
+        v._setup_global(_config_fixture)
+        assert v.name == "test"
+        assert v.region is None
+        assert v.crs.to_epsg() == 4326
+        assert v.input_path == _test_dir / "static/network"
+        assert v.output_path == _test_dir / "output"
+
+    def test_get_network_opt(self, _config_fixture):
+        v = VectorNetworkWrapper(_config_fixture)
+        network_dict = v._get_network_opt(_config_fixture["network"])
+        assert network_dict["files"][0].is_file()
+        assert network_dict["file_id"] == "fid"
+        assert network_dict["file_crs"].to_epsg() == 4326
+        assert network_dict["is_directed"] is False
+
+    def test_read_vector_to_project_region_and_crs(self, _config_fixture):
+        v = VectorNetworkWrapper(_config_fixture)
+        files = v.network_dict["files"]
+        file_crs = v.network_dict["file_crs"]
+        vector = v._read_vector_to_project_region_and_crs(files, file_crs)
+        assert isinstance(vector, gpd.GeoDataFrame)
+
+    def test_read_vector_to_project_region_and_crs_with_region(self, _config_fixture):
+        _config_fixture["project"]["region"] = _test_dir / "_test_polygon.geojson"
+        v = VectorNetworkWrapper(_config_fixture)
+        files = v.network_dict["files"]
+        file_crs = v.network_dict["file_crs"]
+        vector = v._read_vector_to_project_region_and_crs(files, file_crs)
+        assert vector.crs == v.region.crs
+        assert v.region.covers(vector.unary_union).all()
+
+    def test_setup_network_from_vector(self, _config_fixture):
+        v = VectorNetworkWrapper(_config_fixture)
+        graph, edges = v.setup_network_from_vector()
+        assert isinstance(graph, nx.MultiGraph)
+        assert isinstance(edges, gpd.GeoDataFrame)
+
+    def test_setup_network_from_vector_with_region(self, _config_fixture):
+        _config_fixture["project"]["region"] = _test_dir / "_test_polygon.geojson"
+        v = VectorNetworkWrapper(_config_fixture)
+        graph, edges = v.setup_network_from_vector()
+        assert isinstance(graph, nx.MultiGraph)
+        assert isinstance(edges, gpd.GeoDataFrame)
+
+    def test_clean_vector(self, lines_gdf):
+        gdf = VectorNetworkWrapper.clean_vector(lines_gdf)
+        assert all(gdf.reset_index(drop=True).geom_equals(lines_gdf))
+
```

Comment:
I don't fully understand what this test is meant to verify as you are doing post-processing of the resulting dataframe.
You can generate expected data at the beginning so that it's clear what the clean_vector result should look like.

Reply:
Because the cleanup_vector function contains only one step that is explode_and_deduplicated_lines, this test checks if the results are equal.